### PR TITLE
Add support for Circom in `sindri init`

### PR DESCRIPTION
**Pull request by sangaline**
This adds support for Circom circuits in the `sindri init` with a basic template scaffold checking that `X === Y`. There are also a few bug fixes that slipped in. A notable thing is that I pulled in `circomlib` as a dependency in the scaffolded project because I think people are likely to want it. Previously we had been excluding files from `.gitignore` when deploying, but I think it makes sense for `node_modules` to be in `.gitignore` but also be part of the deploy here, so I changed that behavior to only check `.sindriignore` files.

For testing this, you can run:

```bash
# Login on production.
sindr login

# Select the "Circom" circuit type when you do this.
sindri init circom-circuit

cd circom-circuit
sindri deploy
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,6 @@
     "url": "git+https://github.com/Sindri-Labs/sindri-js.git"
   },
   "author": "Sindri Labs",
-  "license": "BSD-2-Clause",
```
Comment (sangaline) [inline package.json:41]: Unrelated, but this was duplicated.

File: src/cli/init.ts [hunk 2]
```diff
@@ -74,13 +74,58 @@ export const initCommand = new Command()
...
+        witnessCompiler,
+      });
+    } else if (circuitType === "gnark") {
+      // Gnark.
       const packageName = await input({
         message: "Go Package Name:",
         default: circuitName
           .replace(/[^a-zA-Z0-9]/g, "")
-          .replace(/^[^a-z]*/, ""),
+          .replace(/^[^a-z]*/g, ""),
```
Comment (sangaline) [inline src/cli/init.ts:128]: Also unrelated, but we need to remove all of these characters at the start of the Gnark package names when there are more than one.

File: src/cli/deploy.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const deployCommand = new Command()
     logger.debug(`Found "sindri.json" at "${sindriJsonPath}".`);
     const rootDirectory = path.dirname(sindriJsonPath);
     logger.debug(`Changing current directory to "${rootDirectory}".`);
-    process.chdir(directoryPath);
+    process.chdir(rootDirectory);
```
Comment (sangaline) [inline src/cli/deploy.ts:38]: Bug fix.

File: templates/circom/example-input.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "X": 1,
+    "Y": 1,
+    "isEqual": 1
```
Comment (katiemckeon) [inline templates/circom/example-input.json:4]: Circom (unlike Gnark) does not include output signals as the input to the circuit.  For this circuit, the `public` field sent back from the API for the proof detail includes "isEqual" as the last entry
```suggestion
    "Y": 1
```

File: templates/circom/circuit.circom [hunk 5]
```diff
@@ -0,0 +1,23 @@
+pragma circom 2.0.8;
+
+// You can include files from `circomlib` like this.
+include "node_modules/circomlib/circuits/comparators.circom";
+
+template templateTemplateName() {
```
Comment (katiemckeon) [inline templates/circom/circuit.circom:6]: I tried the `init` command before reading any of the code, and I found the "Circom template name" question opaque.  Circom code can have many template definitions linking together in arbitrary ways.  So when I answered that question I had a feeling I was signing up for something more than I understood.  My suggestion is to just eliminate that question from the I/O and change this from a templated variable to: 
```suggestion
template Main() {
```

**Review by katiemckeon — COMMENTED** [review]
Nice work! The circuit does compile and it does what you expect it to.  I was able to create proofs (from production API) that match the expected output for a handful of (X,Y) pairs, after revising the `example-input.json`.  

Also the `deploy` command was smooth sailing!  So just a few circom tweaks and I would be satisfied with this.

File: templates/circom/circuit.circom (hunk 5 above)
Comment (sangaline) [reply templates/circom/circuit.circom]: Sounds good. I made all of the related changes in [65d5995](https://github.com/Sindri-Labs/sindri-js/pull/24/commits/65d59952ea5b3e22ca443c029d0b686a8230cb45).

**Review by katiemckeon — APPROVED** [review]
LGTM

**Pull request merged**